crates/threshold-signature-server/src/helpers/launch.rs

```diff
@@ -406,45 +281,69 @@ pub async fn check_node_prerequisites(url: &str, account_id: &str) {
         Ok((api, rpc))
     };
 
-    // Note: By default this will wait 15 minutes before it stops retry attempts.
-    let backoff = backoff::ExponentialBackoff::default();
-    match backoff::future::retry(backoff, connect_to_substrate_node).await {
+    // Never give up trying to connect
+    let backoff = backoff::ExponentialBackoff { max_elapsed_time: None, ..Default::default() };
+
+    match backoff::future::retry(backoff.clone(), connect_to_substrate_node).await {
         Ok((api, rpc)) => {
             tracing::info!("Sucessfully connected to Substrate node!");
 
             tracing::info!("Checking balance of threshold server AccountId `{}`", &account_id);
-            let balance_query = crate::validator::api::check_balance_for_fees(
-                &api,
-                &rpc,
-                account_id.to_string(),
-                entropy_shared::MIN_BALANCE,
-            )
-            .await
-            .map_err(|_| Err::<bool, String>("Failed to get balance of account.".to_string()));
-
-            match balance_query {
-                Ok(has_minimum_balance) => {
-                    if has_minimum_balance {
-                        tracing::info!(
-                            "The account `{}` has enough funds for submitting extrinsics.",
-                            &account_id
-                        )
-                    } else {
-                        tracing::warn!(
-                            "The account `{}` does not meet the minimum balance of `{}`",
-                            &account_id,
-                            entropy_shared::MIN_BALANCE,
-                        )
-                    }
-                },
-                Err(_) => {
-                    tracing::warn!("Unable to query the account balance of `{}`", &account_id)
-                },
+
+            let balance_query = || async {
+                let has_minimum_balance = crate::validator::api::check_balance_for_fees(
+                    &api,
+                    &rpc,
+                    account_id.to_ss58check().to_string(),
+                    entropy_shared::MIN_BALANCE,
+                )
+                .await
+                .map_err(|e| {
+                    tracing::warn!("Account: {} {}", &account_id, e);
+                    e.to_string()
+                })?;
+                if !has_minimum_balance {
+                    Err("Minimum balance not met".to_string())?
+                }
+                Ok(())
+            };
+
+            if let Err(error) = backoff::future::retry(backoff.clone(), balance_query).await {
+                tracing::error!("This should never happen because backoff has no permanent errors or maximum timeout: {error}");
```

Comment:
```suggestion
                tracing::error!(
                    "This should never happen because backoff has no permanent errors or maximum \
                    timeout: {error}"
                );
```

The long lines are a bit hard to read